core-bundle/src/Resources/contao/classes/Versions.php

```diff
@@ -817,20 +817,13 @@ protected function getEditUrl()
 	 */
 	protected function getUsername()
 	{
-		if ($this->strUsername !== null)
-		{
-			return $this->strUsername;
-		}
-
-		$this->import(BackendUser::class, 'User');
-
-		return $this->User->username;
+		return $this->strUsername ?? System::getContainer()->get('security.helper')->getUser()->getUserIdentifier();
```

Comment:
@contao/developers Would you prefer checking if there is a user at all and returning an empty string or `N/A` if there is not?

Reply:
I think we should throw an exception if there is no user.

If using the Versions class in the frontend you should IMO either call `create(blnHideUser: true)` or set the id and name beforehand.

Reply:
Ah, no, sorry. I cannot agree with that, because there are cases where the `Versions` class is used in a helper class that is run in both the back end and the front end. And I would like the class to automatically handle both cases without me having to check the scope and calling the `create()` method with either `true` or `false`.